crates/db_schema/src/impls/custom_emoji.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{
+  schema::{
+    custom_emoji::dsl::custom_emoji,
+    custom_emoji_keyword::dsl::{custom_emoji_id, custom_emoji_keyword},
+  },
+  source::{
+    custom_emoji::{CustomEmoji, CustomEmojiInsertForm, CustomEmojiUpdateForm},
+    custom_emoji_keyword::{CustomEmojiKeyword, CustomEmojiKeywordInsertForm},
+  },
+  utils::{get_conn, DbPool},
+};
+use diesel::{dsl::insert_into, result::Error, ExpressionMethods, QueryDsl};
+use diesel_async::RunQueryDsl;
+
+impl CustomEmoji {
+  pub async fn create(pool: &DbPool, form: &CustomEmojiInsertForm) -> Result<Self, Error> {
+    let conn = &mut get_conn(pool).await?;
+    insert_into(custom_emoji)
+      .values(form)
+      .get_result::<Self>(conn)
+      .await
+  }
+  pub async fn update(
+    pool: &DbPool,
+    emoji_id: i32,
+    form: &CustomEmojiUpdateForm,
+  ) -> Result<Self, Error> {
+    let conn = &mut get_conn(pool).await?;
+    diesel::update(custom_emoji.find(emoji_id))
+      .set(form)
+      .get_result::<Self>(conn)
+      .await
+  }
+  pub async fn delete(pool: &DbPool, emoji_id: i32) -> Result<usize, Error> {
+    let conn = &mut get_conn(pool).await?;
+    diesel::delete(custom_emoji.find(emoji_id))
+      .execute(conn)
+      .await?;
+    diesel::delete(custom_emoji_keyword.filter(custom_emoji_id.eq(emoji_id)))
```

Comment:
You can remove this. As long as the foreign key has cascading deletes, it should automatically delete any of the keyword rows since that emoji_id is now gone.

Reply:
Fixed